app/models/mixins/inter_region_api_method_relay.rb

```diff
@@ -0,0 +1,87 @@
+module InterRegionApiMethodRelay
+  def self.extended(klass)
+    unless klass.const_defined?("InstanceMethodRelay")
+      instance_relay = klass.const_set("InstanceMethodRelay", Module.new)
+      klass.prepend(instance_relay)
+    end
+
+    unless klass.const_defined?("ClassMethodRelay")
+      class_relay = klass.const_set("ClassMethodRelay", Module.new)
+      klass.singleton_class.prepend(class_relay)
+    end
+  end
+
+  def api_relay_method(method, action = method)
+    relay = const_get("InstanceMethodRelay")
+    collection_name = collection_for_class
+
+    relay.class_eval do
+      define_method(method) do |*meth_args, &meth_block|
+        api_args = yield(*meth_args) if block_given?
+
+        if in_current_region?
+          super(*meth_args, &meth_block)
+        else
+          InterRegionApiMethodRelay.exec_api_call(region_number, collection_name, action, api_args) do
```

Comment:
It seems like we would need to check the API response to verify that it was successful and raise an error to the caller if it failed for some reason.